DirectProgramming/DPC++/StructuredGrids/1d_HeatTransfer/src/1d_HeatTransfer.cpp

```diff
@@ -60,114 +60,177 @@ void Usage(string programName) {
   cout << " i : Number of timesteps \n";
 }
 
-//************************************
-// Function description: initialize the array.
-//************************************
-void Initialize(float* array, unsigned int num, unsigned int idx) {
-  for (unsigned int i = idx; i < num; i++) array[i] = 0.0f;
+//
+// Initialize the temperature arrays
+//
+void Initialize(float* arr, float* arr_next, unsigned int num) {
+  arr[0] = arr_next[0] = initial_temperature;
+  for (unsigned int i = 0; i < num; i++)
+    arr[i] = arr_next[i] = 0.0f;
 }
 
-//************************************
-// Function description: compute the heat in the device (in parallel).
-//************************************
-float* ComputeHeatDeviceParallel(float* arr, float* arr_next, float C,
-                                 unsigned int num_p, unsigned int num_iter,
-                                 float temp) {
-  unsigned int i;
+//
+// Compare host and device results
+//
+bool CompareResults(string prefix, float* device_results, float* host_results,
+                    unsigned int num_point, float C) {
+  string path = prefix + "_error_diff.txt";
+  float delta = 0.001f;
+  float difference = 0.00f;
+  double norm2 = 0;
+  bool err = false;
 
-  try {
-    // Define the device queue
-    queue q = default_selector{};
```

Comment:
Should follow dpc_common.hpp pattern from async exceptions.

Reply:
fixed.
Note that dpc_common.hpp is not part of open source dpc++ so the samples are not portable.